sound/soc/sof/ipc4-topology.c

```diff
@@ -834,26 +834,53 @@ static int sof_ipc4_widget_setup_comp_process(struct snd_sof_widget *swidget)
 	if (ret)
 		goto err;
 
-	cfg_size = sizeof(struct sof_ipc4_base_module_cfg);
+	ret = sof_ipc4_widget_setup_msg(swidget, &process->msg);
+	if (ret)
+		goto err;
+
+	/* parse process init module payload config type from module info */
+	fw_module = swidget->module_info;
+	process->init_config =
+		fw_module->man4_module_entry.type & SOF_IPC4_MODULE_INIT_CONFIG_MASK;
+	process->init_config >>= SOF_IPC4_MODULE_INIT_CONFIG_SHIFT;
```

Comment:
would this work instead:

process->init_config = FIELD_GET(SOF_IPC4_MODULE_INIT_CONFIG_MASK, fw_module->man4_module_entry.type)

that's what I used for HDAudio multi-link parsing

Reply:
nice, thanks for the suggestion. changed now.